client/verta/verta/registry/_validate_input.py

```diff
@@ -0,0 +1,34 @@
+import functools
+import json
+
+import jsonschema
+
+# for use like: `if getattr(model.predict, _DECORATED_FLAG, False)`
+_DECORATED_FLAG = "_verta_validate_input"
+_INPUT_SCHEMA_PATH = "/app/model_schema.json"
+
+
+def validate_input(f):
+    """Decorator to validate prediction input against previously provided schema.
+
+    """
+
+    @functools.wraps(f)
+    def wrapper(self, *args, **kwargs):
+        # accepts dict input
+        # TODO: validate length of args and type
+        prediction_input = args[0]
+        try:
```

Comment:
A `try` block in a `try-except` should contain the minimal amount of code that's relevant to the error, as a general practice to avoid accidentally capturing errors from other code (like what if `jsonschema.validate()` or `f()` raises `FileNotFoundError`s? but it's not what we're interested in here).

So something like

```python
try:
    with open(_INPUT_SCHEMA_PATH, 'r') as file:
        schema = json.load(file)
except FileNotFoundError:
    ...
```

Reply:
I forgot this is allowed in python "scoping"!